sklearn/feature_selection/_rfe.py

```diff
@@ -811,3 +925,42 @@ def fit(self, X, y, groups=None):
             "n_features": step_n_features_rev,
         }
         return self
+
+    def get_metadata_routing(self):
+        """Get metadata routing of this object.
+
+        Please check :ref:`User Guide <metadata_routing>` on how the routing
+        mechanism works.
+
+        .. versionadded:: 1.6
+
+        Returns
+        -------
+        routing : MetadataRouter
+            A :class:`~sklearn.utils.metadata_routing.MetadataRouter` encapsulating
+            routing information.
+        """
+        router = MetadataRouter(owner=self.__class__.__name__)
+        router.add(
+            estimator=self.estimator,
+            method_mapping=MethodMapping()
+            .add(caller="fit", callee="fit")
+            .add(caller="predict", callee="predict")
+            .add(caller="score", callee="score"),
+        )
+        router.add(
+            splitter=check_cv(self.cv),
+            method_mapping=MethodMapping().add(
+                caller="fit",
+                callee="split",
+            ),
+        )
+        router.add(
+            scorer=self._get_scorer(),
+            method_mapping=MethodMapping().add(caller="fit", callee="score"),
+        )
+
+        return router
+
+    def _get_scorer(self):
+        return check_scoring(self.estimator, scoring=self.scoring)
```

Comment:
I think this is safer to use `get_scorer`:


```suggestion
        if self.scoring is None:
            scoring = "accuracy" if is_classifier(self.estimator) else "r2"
        else:
            scoring = self.scoring
        return get_scorer(scoring)
```

My issue with `check_scoring` is triggered when `self.scoring is None`. Indeed, we are going to delegate to the `estimator.score` method and thus we don't return a scorer.

I'm wondering if there is any side effect when returning the `estimator.score` that does not have the `score_params` in the signature. I assume that we should have the same error than when calling a `scorer` with an estimator that does not support it.

Reply:
Yeah _PassThroughScorer breaks on 3 tests because of recursion depth exceeded meaning it gets stuck in an endless loop. The code you suggested seems fine, we just have to tweak the MockClassifier defined in the tests a bit to make the tests pass. Could you have a look at the latest changes to see if they look alright and then I can add the additional changelog.